### DISCUSSION TITLE
Regarding feature: "Ask user for a yes/no prompt before installing/removing a package"

### DISCUSSION THREAD

**Discussion by ss141309** [SubconsciousCompute/meta-package-manager#62] [General]
The way we currently handle writing to stdout, is not compatible with interactivity in the cli.
We first collect the output from stdout and then print the lines after prefixing it with "MPM", this messes up the stdin in ways, for eg. the stdin will open before printing the Yes/no line or the line will be missing altogether.

Therefore if we want to implement the feature, I propose the following:
We can pass an interactivity flag to the program to turn on this feature.
When the flag is passed, we do not prefix the output of the stdout and print them as it is, instead we can print a banner at the top indicating that `mpm` is being run.

The output will look something like this:
![image](https://github.com/user-attachments/assets/0e31ccdc-1a63-4438-bcf7-5da11b5618d1)

How does this sound?

**Comment by ss141309** [comment]
@dilawar how does this proposal sound?

**Reply by dilawar** [reply]
Sounds good @ss141309 . Go ahead and implement it. But leave a detailed note why you are manking these changes in the right place.

If these changes are not easily irreversible, please think about them again.
